resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -156,6 +160,14 @@ private[yarn] class YarnAllocator(
   private[yarn] val containerPlacementStrategy =
     new LocalityPreferredContainerPlacementStrategy(sparkConf, conf, resource, resolver)
 
+  def createResourceCapability: Resource = {
```

Comment:
looks like this is only used to initialize `resource`, you could just inline it.  (at least should be `private`)

Reply:
good point, fixed